core/dbt/contracts/graph/manifest.py

```diff
@@ -1008,6 +1008,7 @@ def merge_from_artifact(
         adapter,
         other: "WritableManifest",
         selected: AbstractSet[UniqueID],
+        favor_state: bool,
```

Comment:
It looks like this line is causing the test failures as `CompileTask.defer_to_manifest` in `task/compile.py` isn't passing the new arg. Try:
```suggestion
        favor_state: bool = False,
```